test/physics/base/PhysicsStepUtils.test.cc

```diff
@@ -136,4 +136,141 @@ TEST_F(PhysicsStepUtilsTest, calc_tabulated_physics_step)
+                              phys_state.ref(),
+                              par_init.particle_id,
+                              mat_init.material_id,
+                              ThreadId{0});
+
+        phys.interaction_mfp(1);
+
+        real_type step
+            = celeritas::calc_tabulated_physics_step(material, particle, phys);
+        EXPECT_SOFT_EQ(4.1595999999999984, step);
+         // Testing cheat.
+        PhysicsTrackView::PhysicsStatePointers state_shortcut(phys_state.ref());
+        state_shortcut.state[ThreadId{0}].interaction_mfp = 0;
+
+        auto result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 1);
+        EXPECT_EQ(result.model.get(), 5);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 1);
+        EXPECT_EQ(result.model.get(), 5);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 1);
+        EXPECT_EQ(result.model.get(), 5);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 0);
+        EXPECT_EQ(result.model.get(), 1);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 2);
+        EXPECT_EQ(result.model.get(), 8);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 1);
+        EXPECT_EQ(result.model.get(), 5);
+        ++num_samples;
+
+        result = select_process_and_model(particle, phys, this->rng());
+        EXPECT_EQ(result.ppid.get(), 0);
+        EXPECT_EQ(result.model.get(), 1);
+        ++num_samples;
```

Comment:
Instead of the copy-pasted code, can you do:
```c++
vector<size_type> selected_ppid(20);
vector<size_type> selected_mode(selected_ppid.size());
for (auto i : range(selected_ppid.size()))
{
    auto result = ...;
    selected_ppid[i] = result.ppid.unchecked_get();
    selected_model[i] = result.model.unchecked_get();
}
PRINT_EXPECTED(selected_ppid);
PRINT_EXPECTED(selected_model);
EXPECT_VEC_EQ(...);
```